build_functions/CMakeLists.txt

```diff
@@ -604,8 +604,8 @@ macro(set_default_build_options)
         string(REGEX REPLACE "/W[1-3]" "/W4" CMAKE_C_FLAGS "${CMAKE_C_FLAGS}")
         
     elseif(UNIX) #LINUX OR APPLE
-        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror")
-        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Werror")
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror -g")
+        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Werror -g")
```

Comment:
We could also change the build type in the script to be "-DCMAKE_BUILD_TYPE=Debug" instead of defaulting to Release (since the Makefile generator bakes in the type unlike the VS one). But I guess this means we will always use essentially RelWithDebInfo on Linux so maybe that's acceptable.

Reply:
If you add the -g here, there will be no way to override it, shouldn't this be optional?  Maybe default it, but make it so we can remove it or add to it.

Reply:
We could. But maybe "when time comes".

---
In reply to: [516905216](https://github.com/Azure/c-build-tools/pull/46#discussion_r516905216) [](ancestors = 516905216)

Reply:
Also "when time comes".

---
In reply to: [516934619](https://github.com/Azure/c-build-tools/pull/46#discussion_r516934619) [](ancestors = 516934619)